src/kernelspace/exec_monitor.c

```diff
@@ -25,14 +25,42 @@ struct {
 } ringbuf SEC(".maps");
 
 long ringbuffer_flags = 0;
+int max_n_proc;
+int n_monitored_proc;
+
+int filter_by_ppid;
+int ppid_list[10];
+int ppid_list_size;
 
 SEC("lsm/bprm_committed_creds")
 void BPF_PROG(exec_audit, struct linux_binprm *bprm)
 {
+	if (max_n_proc != 0 && n_monitored_proc == max_n_proc)
+		return;
+
 	long pid_tgid;
+	int ppid, selected_ppid = 0;
 	struct process_info *process;
 	struct task_struct *current_task;
 
+	// Get the parent pid
+	current_task = (struct task_struct *)bpf_get_current_task();
+	ppid = (int)BPF_CORE_READ(current_task, real_parent, pid);
+
+	if (filter_by_ppid) {
+		for (int i = 0; i < 10; i++) {
+			if (ppid == ppid_list[i]) {
+				selected_ppid = 1;
+				break;
+			}
+			if (i == ppid_list_size)
```

Comment:
I don't think this does what you want.
i evolves in the [0, 9] space.
ppid_list_size evolves in the [1, 10] space.

If you take the simple case of a ppid_list_size = 1, your loop will iterate on i = 0, try to check if ppid = ppid_list[0], verify that i != ppid_list_size, then iterate again and check that ppid == ppid_list[1] which is already out of bound. (but will silently work since ppid_list is in bss ans you will have ppid_list[1] to zero.

If you move this check to the for loop, you avoid the issue.